sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.confirmEnrollment.e2e-spec.ts

```diff
@@ -0,0 +1,298 @@
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { ClientTypeBaseRoute } from "../../../../types";
+
+describe(`${ClientTypeBaseRoute.Institution}-ConfirmationOfEnrollmentInstitutionsController(e2e)-confirmEnrollment`, () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let applicationRepo: Repository<Application>;
+  let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+  let offeringRepo: Repository<EducationProgramOffering>;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+    offeringRepo = dataSource.getRepository(EducationProgramOffering);
+
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+  });
+
+  it("Should allow the COE confirmation when the application is on Enrolment status and all the conditions are fulfilled.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationCOE(appDataSource, {
+      institution: collegeC,
+      institutionLocation: collegeCLocation,
+    });
+    application.applicationStatus = ApplicationStatus.Enrolment;
+    await applicationRepo.save(application);
+    const [fistDisbursementSchedule] =
+      application.currentAssessment.disbursementSchedules;
+    const endpoint = `/institutions/location/${collegeCLocation.id}/confirmation-of-enrollment/disbursement-schedule/${fistDisbursementSchedule.id}/confirm`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send({ tuitionRemittanceAmount: 1 })
+      .auth(
+        await getInstitutionToken(InstitutionTokenTypes.CollegeCUser),
+        BEARER_AUTH_TYPE,
+      )
+      .expect(HttpStatus.OK);
+    // Check if the application was updated as expected.
+    const updatedApplication = await applicationRepo.findOne({
+      select: { applicationStatus: true },
+      where: { id: application.id },
+    });
+    expect(updatedApplication.applicationStatus).toBe(
+      ApplicationStatus.Completed,
+    );
+  });
+
+  it("Should allow the COE confirmation when the application is on Completed status and all the conditions are fulfilled.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationCOE(appDataSource, {
+      institution: collegeC,
+      institutionLocation: collegeCLocation,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
```

Comment:
First disbursement being pending on completed application can happen during reassessment. Nice thought process. 👍

Reply:
For this I was just trying to satisfy the query 😉